### PR TITLE
Validate collected subdomains

### PR DESCRIPTION
**Pull request by ErebusZ**

### CODE REVIEW

File: agent/subfinder_agent.py [hunk 1]
```diff
@@ -59,7 +60,13 @@ def process(self, message: m.Message) -> None:
                     sub_domains = sub_domains[: self.args.get("max_subdomains")]
 
                 for sub in sub_domains:
-                    self.emit(selector="v3.asset.domain_name", data={"name": sub})
+                    try:
+                        if validators.domain(sub) is True:
```
Comment (oussamaessaji) [inline agent/subfinder_agent.py:64]: `validators.domain("invalid.com")` will return true is this the right behavior?

**Review by amine3 — APPROVED** [review]

File: agent/subfinder_agent.py (hunk 1 above)
Comment (ErebusZ) [reply agent/subfinder_agent.py:64]: yes, `True` if domain is valid and raises `ValidationError` if not

**Review by mohsinenar — APPROVED** [review]

File: tests/subfinder_agent_test.py [hunk 2]
```diff
@@ -72,3 +72,26 @@ def testAgentSubfinder_whenMaxSubDomainsSet_emitsBackFindings(
     assert agent_mock[0].selector == "v3.asset.domain_name", (
         agent_mock[0].data["name"] == "subdomain1"
     )
+
+
+def testAgentSubfinder_whenSubDomainIsInvalid_theSubDomainIsNotReported(
+    subfinder_agent, agent_persist_mock, agent_mock, mocker
+):
+    """Unittest for Agent Subfinder. When it receives an invalid subdomain from subfinder
```
Comment (rabsondev) [inline tests/subfinder_agent_test.py:80]:
```suggestion
    """Unit test for Agent Subfinder. When it receives an invalid subdomain from subfinder
```

File: agent/subfinder_agent.py [hunk 3]
```diff
@@ -59,7 +60,13 @@ def process(self, message: m.Message) -> None:
...
+                            self.emit(
+                                selector="v3.asset.domain_name", data={"name": sub}
+                            )
+                    except validators.ValidationError:
+                        logger.info("Got an invalid subdomain: %s", sub)
```
Comment (deadly-panda) [inline agent/subfinder_agent.py:69]:
```suggestion
                        logger.warning("Got an invalid subdomain: %s", sub)
```

**Comment by 3asm** [comment]
Why has this PR been re-opened. Subfinder MUST NOT filter any of its domains.

**Pull request closed without merging**